x/app-router/types/parser.go

```diff
@@ -18,30 +18,33 @@ type ParsedReceiver struct {
 // need args: Creator, HostDenom, Amount
 func ParseReceiverData(receiverData string) (*ParsedReceiver, error) {
 	parts := strings.Split(receiverData, "|")
-	addressPart := parts[0]
 
-	// Standard address
-	if len(parts) == 1 && addressPart != "" {
+	switch len(parts) {
+	case 1:
 		return &ParsedReceiver{
 			ShouldLiquidStake: false,
 		}, nil
-	}
+	case 2:
+		addressPart := parts[0]
+		functionPart := parts[1]
+
+		// verify the rightmost field is stakeibc/liquidstake
+		if functionPart != "stakeibc/liquidstake" {
```

Comment:
As commented above, should we use the amino types here? `stakeibc/LiquidStake`

Reply:
Done here https://github.com/Stride-Labs/stride/pull/432/commits/d17ffb6028a32eba576bf75d10c226dfc539278e